sqlx-postgres/src/transaction.rs

```diff
@@ -26,6 +27,24 @@ impl TransactionManager for PgTransactionManager {
         })
     }
 
+    fn begin_with<'a, S>(
+        conn: &'a mut <Self::Database as Database>::Connection,
+        sql: S,
+    ) -> BoxFuture<'a, Result<(), Error>>
+    where
+        S: Into<Cow<'static, str>> + Send + 'a,
+    {
+        Box::pin(async move {
+            let rollback = Rollback::new(conn);
```

Comment:
This should check and return an error if `conn.transaction_depth > 0` because it doesn't make sense to allow this to create and release manual savepoints when a transaction is already in progress; that's just going to break things.